src/main/java/com/atlassian/db/replica/api/AuroraMultiReplicaConsistency.java

```diff
@@ -32,6 +39,7 @@ public boolean isConsistent(Supplier<Connection> replicaSupplier) {
     public static final class Builder {
         private ReplicaConsistency replicaConsistency;
         private AuroraConnectionDetails auroraConnectionDetails;
+        private SuppliedCache<Collection<Database>> discoveredReplicasCache = new NoCacheSuppliedCache<>();
```

Comment:
What's the cache for?

Reply:
We don't need to fetch existing replicas every millisecond and in parallel, but with certain throttling, like every 100ms or maybe 1s. Replicas are scaled up and down dynamically and we have to balance between how fast we discover the change vs what overhead we are adding.

Reply:
Can you add a JavaDoc?